text/0000-serialize-deserialize.md

```diff
@@ -0,0 +1,174 @@
+* Start Date: 2018-08-01
+* RFC PR: https://github.com/fusionjs/rfcs/pull/8
+
+# Summary
+
+This RFC describes a process for standardizing on an API for serializing and deserializing state between server and browser plugins.
+
+# Basic example
+
+```js
+import {SharedStateToken, createPlugin} from 'fusion-core';
+
+export default createPlugin({
+  deps: {
+    SharedState: SharedStateToken,
+  },
+  middleware: ({SharedState}) => {
+    const {serialize, deserialize} = SharedState('some-id');
+    return (ctx, next) => {
+      if (__BROWSER__) {
+        // load and parse serialized data from dom node
+        const data = deserialize();
+      } else {
+        // serialize json data into body
+        serialize(ctx, {
+          some: 'json',
+        });
+      }
+      return next();
+    };
+  },
+});
+
+// in tests
+import {SharedStateToken, MockSharedState} from 'fusion-core';
+app.register(SharedStateToken, MockSharedState({
+  'some-id': {
+    some: 'json'
+  },
+  'other-data': {}
```

Comment:
How would this work with custom serialization?

Reply:
This doesn't actually change the serialization, only the deserialization. So whether it is custom or default, this will return whatever the value is in the map. You could put an immutable-js object here directly